From 88df95bd7f1b5eb1af96a37582728b86336b722a Mon Sep 17 00:00:00 2001
From: Morten Brekkevold <morten.brekkevold@uninett.no>
Date: Mon, 18 Feb 2019 11:39:11 +0100
Subject: [PATCH] Don't mix date and datetime objects in business reports.

If you do, there will be this error:

> TypeError: can't compare datetime.datetime to datetime.date

Fixes #1857
---
 python/nav/web/business/reportengine.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/nav/web/business/reportengine.py b/python/nav/web/business/reportengine.py
index 6ae34d7a5..c14d69d17 100755
--- a/python/nav/web/business/reportengine.py
+++ b/python/nav/web/business/reportengine.py
@@ -76,7 +76,7 @@ def build_report(period, report_type, exclude_maintenance=False):
 
 def build_context(period, report_type, exclude_maintenance=False):
     """Builds a context for the given subscription"""
-    midnight = date.today()
+    midnight = datetime.combine(date.today(), datetime.min.time())
     start, end = get_last_interval(midnight, period)
     lookup = {
         ReportSubscription.DEVICE: utils.get_netbox_records,
-- 
2.17.1

